### PR TITLE
Log stdout and stderr when a command is not running as expected.

### PR DESCRIPTION
**Pull request by guillaumerose**
The intent is to add more details when a command fails.

Example: https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/origin-ci-test/pr-logs/pull/code-ready_crc/1297/pull-ci-code-ready-crc-master-e2e-crc/1274295022597967872
```
  Scenario: Push local image to OpenShift image registry # features/story_registry.feature:24
    When executing "sudo podman push hello:test default-route-openshift-image-registry.apps-crc.testing/testproj-img/hello:test --tls-verify=false" succeeds # features/story_registry.feature:28
      Error: command 'sudo podman push hello:test default-route-openshift-image-registry.apps-crc.testing/testproj-img/hello:test --tls-verify=false', expected to succeed, exited with exit code: 125
```

### CODE REVIEW

**Review by gbraad — APPROVED** [review]

**Comment by gbraad** [comment]
@jsliacan Have a look. Adding you as the reviewer

The actual output is logged already as part of: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/code-ready_crc/1342/pull-ci-code-ready-crc-master-e2e-crc/1276630845381152768/artifacts/e2e-crc/gather-crc/test-results/integration_2020-6-26_21-51-13.log
```
22:41:11.484969    bash: sudo podman login -u kubeadmin -p $(oc whoami -t) default-route-openshift-image-registry.apps-crc.testing --tls-verify=false
22:41:12.785172  stderr: Error: error authenticating creds for "default-route-openshift-image-registry.apps-crc.testing": unexpected http code: 502 (Bad Gateway), URL: https://default-route-openshift-image-registry.apps-crc.testing/openshift/token?account=kubeadmin
22:41:12.789978  stdout: exitCodeOfLastCommandInShell=125
```

**Comment by jsliacan** [comment]
LGTM, agree with the intent, it'll save us time going through artifacts. I hope it won't be too wordy for some errors.

**Comment by gbraad** [comment]
Might this record or print some PII or credentials in the output?

**Comment by guillaumerose** [comment]
Yes but it's already in an other log. It's just more convenient.
